### ISSUE TITLE
Bug - No StackTrace on failed tests

### ISSUE THREAD

**Issue by Faz540** [webdriverio/webdriverio#2711]
This appears to have been raised and closed in the past:
https://github.com/webdriverio/wdio-sync/issues/45
https://github.com/webdriverio/wdio-sync/issues/867

Feel free to clone the below repo and run the basic test.
https://bitbucket.org/Faz540/webdriverio-issue

I've added a 'z' to an element's ID, which obviously doesn't exist but the error that is logged in the console is:
![image](https://user-images.githubusercontent.com/9334128/39362316-f039827a-4a1d-11e8-81dc-b49dfb13827f.png)

I would expect it to display a line number, along with the file that contains the failure.

I'm pretty confident this used to show, I hope I'm just missing something obvious in my config.

**Comment by christian-bromann** [comment]
Feel free to provide a PR with a fix.

**Comment by Faz540** [comment]
This appears to be a regressed issue as this used to show the directory/line number:
![image](https://user-images.githubusercontent.com/9334128/39420321-e69264ae-4c5b-11e8-8cfa-5653e6018508.png)

**Comment by Lukukas** [comment]
I've also noticed this. Been working around it, but would also really like to have the full stack trace appear.

**Comment by Faz540** [comment]
@christian-bromann Did you clone the repo and manage to recreate it in your environment? I had another tester clone the repo, who had a different Node version to me, and they had the same issue.

I'm just confused why it used to work. I've tried downgrading all my node packages to versions that were 6 months+ old and the issue is still present.

**Comment by Faz540** [comment]
Bizarre! I cloned the repo on my home computer, ran the test and the stack trace was shown...

I guess you're right Christian :P 
I just need to learn how to replicate my home environment at my work computer now.

@christian-bromann  Feel free to close issue now

![image](https://user-images.githubusercontent.com/9334128/39943793-11bb173a-555c-11e8-80de-be92b18b53ad.png)

**Comment by christian-bromann** [comment]
@Faz540 I had not time yet to check this use case. Feel free to look into it but I try to fix this issue with v5

**Comment by christian-bromann** [comment]
Hey,

we update the core architecture of WebdriverIO and with that a lot of implementation details that might make this issue obsolete. Please try with the v5 version of WebdriverIO and verify if this still occurs. If so please open a new issue and we will tackle it asap.

Thanks!

**Comment by klamping** [comment]
After testing this out in v5, I don't this issue is resolved. 

For example, in the 'examples' folder, if I run [the 'form' tests](https://github.com/webdriverio/webdriverio/blob/master/examples/pageobject/specs/form.spec.js) that are part of the page objects examples, and use an invalid selector for the submit button, the error output is: 
```
[chrome 70.0.3538.77 Mac OS X #0-0] 1) auth form should deny access with wrong creds
[chrome 70.0.3538.77 Mac OS X #0-0] Can't call click on element with selector "#loginnotfound button[type=submit]" because element wasn't found
[chrome 70.0.3538.77 Mac OS X #0-0] Error: Can't call click on element with selector "#loginnotfound button[type=submit]" because element wasn't found
[chrome 70.0.3538.77 Mac OS X #0-0]     at Promise.then (/Users/klamping/Sites/webdriverio/packages/webdriverio/src/middlewares.js:48:31)
[chrome 70.0.3538.77 Mac OS X #0-0]     at <anonymous>
```

While I know my test failed, and I know the element that can't be found, I don't know where in the test it failed.

This can be tricky for longer specs, especially when they use page objects. It gets trickier when the error occurred somewhere in the middle of a flow where we're testing elements existing and being removed (e.g. which time checking for an element did it disappear?)

I've tried looking at the webdriverio code for some sort of easy fix, but I think we're really limited by how NodeJS handles stack traces. I don't think the `elementErrorHandler` function inside the `middleware.js` file has any more information about who is calling it.

Maybe this needs to be updated in the test framework repos, but the best I can do there is get the line in the spec file, which doesn't help with identifying where in the page object file the issue is occurring. 

Any ideas on where I can poke around looking for better answers would be much appreciated. Much of my time debugging wdio tests revolve around these stack traces just not having enough information in them.

**Comment by simplyageek** [comment]
Any progress with this? It is really frustrating to debug, not knowing exactly, where the problem in the code is.

**Comment by goodbomb** [comment]
+1

**Comment by IrinaSouth** [comment]
+1

**Comment by WillBrock** [comment]
This should be fixed with #3565
